certificate-rotation.md

```diff
@@ -0,0 +1,113 @@
+There are two parts to certificate rotation: generating the new certificates, and reloading them.
+
+Both have been improved, with the main difference being that we can now reload certificates without restarting a node.
+
+## Order of certificate regeneration
+
+Most of the complication in rotating certificates is proper handling of new CA certificates.
+It's important for all parts of the system (nodes and clients) to have the same CA certificates, or they may not be able to verify each other's client/node certificates.
+
+It is possible to have multiple CA certificates active at the same time, but not node or client certificates. Therefore, CA certificates should be generated and pushed first, followed by new node/client certificates using the new CA.
+
+Let's say we want to regenerate all certificates (CA, client, node), we should do the following:
+1. generate new CA certificate and combine it with the existing one
+1. push combined CA certificate to all nodes and reload certificates
+1. make combined CA available to all clients. We may need to wait some time (days/weeks) for all clients to make use of the new certificate
+1. generate and push new node/client certificates using the new CA certificate
+
+It is recommended to change CA certificates on a completely different schedule. eg: months before.
```

Comment:
yup. Ideally, we want everyone to have the same CA certs. It's easy for nodes since we control that and can push the CA cert to all of them fairly quickly. We may not control how clients access the CA though. If it's all from a shared location, that may be easy, but even then the clients will need to reload the certificate (usually meaning a restart)